v2/ical.NET/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -299,7 +299,7 @@ private void EnforceEvaluationRestrictions(IRecurrencePattern pattern)
                         {
                             break;
                         }
-                        else if (pattern.Until == DateTime.MinValue || candidate <= pattern.Until)
```

Comment:
as you can see, these conditions are redundant with line 305 anyways, but here the candidate is not treated in the UTC timezone, hence the bug, as we're comparing apples to oranges.